modules/gdnative/nativearvr/arvr_interface_gdnative.cpp

```diff
@@ -0,0 +1,286 @@
+
+	return stereo;
+};
+
+bool ARVRInterfaceGDNative::is_initialized() {
+	void *parameters[1];
+	bool initialized;
+
+	ERR_FAIL_COND_V(data == NULL, false);
+
+	parameters[0] = (void *)this;
+	const_cast<GDNative *>(library.ptr())->call_native_raw("arvr_return_bool", "godot_arvr_is_initialized", data, 1, parameters, &initialized);
+
+	return initialized;
+};
+
+bool ARVRInterfaceGDNative::initialize() {
+	void *parameters[1];
+	bool initialized;
+
+	ERR_FAIL_COND_V(data == NULL, false);
+
+	parameters[0] = (void *)this;
+	const_cast<GDNative *>(library.ptr())->call_native_raw("arvr_return_bool", "godot_arvr_initialize", data, 1, parameters, &initialized);
+
+	if (initialized) {
+		// if we successfully initialize our interface and we don't have a primary interface yet, this becomes our primary interface
+
+		ARVRServer *arvr_server = ARVRServer::get_singleton();
+		if ((arvr_server != NULL) && (arvr_server->get_primary_interface() == NULL)) {
+			arvr_server->set_primary_interface(this);
+		};
+	};
+
+	return initialized;
+}
+
+void ARVRInterfaceGDNative::uninitialize() {
+	void *parameters[1];
+
+	ERR_FAIL_COND(data == NULL);
+
+	ARVRServer *arvr_server = ARVRServer::get_singleton();
+	if (arvr_server != NULL) {
+		// Whatever happens, make sure this is no longer our primary interface
+		arvr_server->clear_primary_interface_if(this);
+	}
+
+	parameters[0] = (void *)this;
+	const_cast<GDNative *>(library.ptr())->call_native_raw("arvr_call_method", "godot_arvr_uninitialize", data, 1, parameters, NULL);
+}
+
+Size2 ARVRInterfaceGDNative::get_recommended_render_targetsize() {
+	void *parameters[1];
+	Size2 size;
+
+	ERR_FAIL_COND_V(data == NULL, Size2());
+
+	parameters[0] = (void *)this;
+	const_cast<GDNative *>(library.ptr())->call_native_raw("arvr_return_vector2", "godot_arvr_get_recommended_render_targetsize", data, 1, parameters, &size);
+
+	return size;
+}
+
+Transform ARVRInterfaceGDNative::get_transform_for_eye(ARVRInterface::Eyes p_eye, const Transform &p_cam_transform) {
+	void *parameters[3];
+	Transform ret;
+
+	ERR_FAIL_COND_V(data == NULL, Transform());
+
+	parameters[0] = (void *)this;
+	parameters[1] = (void *)&p_eye;
+	parameters[2] = (void *)&p_cam_transform;
+	const_cast<GDNative *>(library.ptr())->call_native_raw("arvr_return_transform_for_eye", "godot_arvr_get_transform_for_eye", data, 3, parameters, &ret);
+
+	return ret;
+}
+
+CameraMatrix ARVRInterfaceGDNative::get_projection_for_eye(ARVRInterface::Eyes p_eye, real_t p_aspect, real_t p_z_near, real_t p_z_far) {
+	void *parameters[6];
+	CameraMatrix cm;
+
+	ERR_FAIL_COND_V(data == NULL, CameraMatrix());
+
+	parameters[0] = (void *)this;
+	parameters[1] = (void *)cm.matrix;
+	parameters[2] = (void *)&p_eye;
+	parameters[3] = (void *)&p_aspect;
+	parameters[4] = (void *)&p_z_near;
+	parameters[5] = (void *)&p_z_far;
+	const_cast<GDNative *>(library.ptr())->call_native_raw("arvr_call_return_projection_for_eye", "godot_arvr_return_projection_for_eye", data, 6, parameters, NULL);
```

Comment:
The call type name is `arvr_call_return_projection_for_eye`, so I would expect it to return something. You probably modify the contents of `cm.matrix` inside the function, which is totally alright, but then I think `godot_arvr_fill_projection_for_eye` would be a better name for the function inside the library.

Reply:
True, good suggestion, I'll change it